clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -420,6 +420,8 @@ static bool IsSyclMathFunc(unsigned BuiltinID) {
 bool Sema::isKnownGoodSYCLDecl(const Decl *D) {
   if (const FunctionDecl *FD = dyn_cast<FunctionDecl>(D)) {
     const IdentifierInfo *II = FD->getIdentifier();
+    if (FD->getBuiltinID() == Builtin::BIprintf)
+      return true;
```

Comment:
Tagging @AaronBallman, @elizabethandrews and @premanandrao for awareness: this `if` should probably be extended to check if we are compiling for CUDA or not.

The thing is that `printf` built-in as-is won't be properly converted into SPIR-V and will cause JIT compilation errors on OpenCL backends.

Reply:
Changing it to check for CUDA makes sense to me, good catch!